db/change_cache.go

```diff
@@ -695,7 +695,11 @@ func (h *SkippedSequenceQueue) Remove(x uint64) error {
 
 	i := SearchSequenceQueue(*h, x)
 	if i < len(*h) && (*h)[i].seq == x {
-		*h = append((*h)[:i], (*h)[i+1:]...)
+		// GC-friendly removal of skipped sequence entries from the queue.
+		// (https://github.com/golang/go/wiki/SliceTricks)
+		copy((*h)[i:], (*h)[i+1:])
+		(*h)[len(*h)-1] = nil
```

Comment:
I believe it's this line that guarantees there isn't a GC-blocking reference to the old `h[len(h)-1]` element in the slice's backing storage, even if the current slice bounds don't include it.